File: feitos/2929 - Menor da Pilha/Main.java
import java.util.Scanner;

class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        Pilha pilha = new Pilha();

        int n = sc.nextInt();
        for (int i = 0; i < n; i++) {
            String input = sc.next();
            int v = 0;
            if (input.equals("PUSH")) {
                v = sc.nextInt();
                pilha.inserir(v);
            } else if (input.equals("POP")) {
                pilha.remover();
            } else if (input.equals("MIN")) {
                System.out.println(pilha.menor());
            }
        }

        sc.close();
    }
}

class Celula {
    int elemento;
    Celula prox;

    Celula(int x) {
        this.elemento = x;
        this.prox = null;
    }
}

class Pilha {
    Celula topo;

    void inserir(int x) {
        Celula tmp = new Celula(x);
        tmp.prox = topo;
        topo = tmp;
    }

    int remover() {
        if (topo == null) System.out.println("EMPTY");

        int resp = topo.elemento;
        topo = topo.prox;
        
        return resp;
    }

    int menor() {
        if (topo == null) System.out.println("EMPTY");
        Celula min = topo;

        for (Celula i = topo; i != null; i = i.prox) {
            if (i.elemento < min.elemento) min = i;
        }

        return min.elemento;
    }
}

File: feitos/2929 - Menor da Pilha/main.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

typedef struct Celula {
    int elemento;
    struct Celula *prox;
} Celula;

Celula *novaCelula(int x) {
    Celula *nova = (Celula *) malloc (sizeof(Celula));
    nova->elemento = x;
    nova->prox = NULL;

    return nova;
}

Celula *topo = NULL;

void inserir(int x) {
    Celula *tmp = novaCelula(x);
    tmp->prox = topo;
    topo = tmp;
}

int remover() {
    if (topo == NULL) printf("EMPTY\n");

    int resp = topo->elemento;
    topo = topo->prox;

    return resp;
}

int min() {
    if (topo == NULL) printf("EMPTY\n");

    Celula *min = topo;
    for (Celula *i = topo; i != NULL; i = i->prox) {
        if (i->elemento < min->elemento) min = i;
    }

    return min->elemento;
}

int main() {
    int n;
    scanf("%d", &n);

    for (int i = 0; i < n; i++) {
        char input[11];
        int v = 0;
        scanf("%s", input);

        if (strcmp(input, "PUSH") == 0) {
            scanf("%d", &v);
            inserir(v);
        } else if (strcmp(input, "POP") == 0) {
            remover();
        } else if (strcmp(input, "MIN") == 0) {
            printf("%d\n", min());
        }
    }

    while (topo != NULL) {
        Celula *tmp = topo;
        topo = topo->prox;
        free(tmp);
    }

    return 0;
}